src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,380 @@
+                && targetArgument.Parent is ArgumentListSyntax argumentList
+                && argumentList.Parent is SyntaxNode invocationNode) // invocation node could be Invocation Expression, Object Creation, Base Constructor...
+            {
+                // Implicit downcast appears on the argument of invocation node, get all candidate functions and extract potential conversion types 
+                var symbolInfo = semanticModel.GetSymbolInfo(invocationNode, cancellationToken);
+                var candidateSymbols = symbolInfo.CandidateSymbols;
+
+                foreach (var candidateSymbol in candidateSymbols.OfType<IMethodSymbol>())
+                {
+                    if (CanArgumentTypesBeConvertedToParameterTypesTest(
+                        semanticModel, root, argumentList, candidateSymbol.Parameters, targetArgument,
+                        cancellationToken, out var targetArgumentConversionType))
+                    {
+                        mutablePotentialConversionTypes.Add(targetArgumentConversionType);
+                    }
+                }
+
+                // Sort the potential conversion types by inheritance distance, so that
+                // operations are in order and user can choose least specific types(more accurate)
+                SortTypesByInheritanceDistance(semanticModel, targetNodeType, mutablePotentialConversionTypes);
+            }
+
+            using var __ = ArrayBuilder<ITypeSymbol>.GetInstance(out var validPotentialConversionTypes);
+            foreach (var targetNodeConversionType in mutablePotentialConversionTypes)
+            {
+                var commonConversion = semanticModel.Compilation.ClassifyCommonConversion(
+                    targetNodeType, targetNodeConversionType);
+
+                // For cases like object creation expression. for example:
+                // Derived d = [||]new Base();
+                // It is always invalid except the target node has explicit conversion operator or is numeric.
+                if (targetNode.IsKind(SyntaxKind.ObjectCreationExpression)
+                    && !commonConversion.IsUserDefined)
+                {
+                    continue;
+                }
+
+                if (commonConversion.Exists)
+                {
+                    validPotentialConversionTypes.Add(targetNodeConversionType);
+                }
+            }
+
+            // clear up duplicate types
+            potentialConversionTypes = validPotentialConversionTypes.Distinct().ToImmutableArray();
+            return !potentialConversionTypes.IsEmpty;
+
+            static void SortTypesByInheritanceDistance(
+                SemanticModel semanticModel, ITypeSymbol targetNodeType,
+                ArrayBuilder<ITypeSymbol> mutablePotentialConversionTypes)
+            {
+                var comparer = new InheritanceDistanceComparer(semanticModel, targetNodeType);
+                mutablePotentialConversionTypes.Sort(comparer);
```

Comment:
called by only one place.  Feels liek this is can inlined.  i.e. the caller should just do `mutablePotentialConversionTypes.Sort(new InheritanceDistanceComparer(semanticModel, targetNodeType))`